pkg/project/auth/cache.go

```diff
@@ -172,6 +178,29 @@ func (ac *AuthorizationCache) Run(period time.Duration) {
 	go utilwait.Forever(func() { ac.synchronize() }, period)
 }
 
+func (ac *AuthorizationCache) AddWatcher(watcher CacheWatcher) {
+	ac.watcherLock.Lock()
+	defer ac.watcherLock.Unlock()
+
+	ac.watchers = append(ac.watchers, watcher)
+}
+
+func (ac *AuthorizationCache) RemoveWatcher(watcher CacheWatcher) {
+	ac.watcherLock.Lock()
+	defer ac.watcherLock.Unlock()
+
+	// just in case we don't remove anything
+	newWatchers := make([]CacheWatcher, len(ac.watchers), 0)
```

Comment:
> If this is a hot path, some benchmarking won't hurt (this impl. vs maps vs whatever else you may fit here).

I wouldn't expect it to be.  This can be done with an overlapping copy, so he's asking for that.